src/traces/bar/plot.js

```diff
@@ -149,14 +149,13 @@ function plot(gd, plotinfo, cdModule, traceLayer, opts, makeOnCompleteCallback)
             );
 
             // display zeros if line.width > 0
-            if(isBlank && shouldDisplayZeros && helpers.getLineWidth(trace, di) && (isHorizontal ? x1 - x0 === 0 : y1 - y0 === 0)) {
+            if(isBlank && shouldDisplayZeros &&
+                helpers.getLineWidth(trace, di) &&
+                (isHorizontal ? x0 === x1 : y0 === y1)) {
                 isBlank = false;
             }
             di.isBlank = isBlank;
 
-            if(isBlank && isHorizontal) x1 = x0;
-            if(isBlank && !isHorizontal) y1 = y0;
```

Comment:
Those two lines added in https://github.com/plotly/plotly.js/commit/f49828a05f?short_path=73806af#diff-73806afb28fb251e00e170f2218e263a appear to cause some rounding problem. @antoinerg do you recall why they were needed?
Now that they removed the new test added in https://github.com/plotly/plotly.js/pull/4568/commits/7f3fb34fc3300f7670a9d92e0054802efb105ac2 with defined range pass.

Reply:
Hmm right, this may have to do with transitions.

@antoinerg can you send us links to your old bar transition examples?

Reply:
Putting the logic back in https://github.com/plotly/plotly.js/pull/4568/commits/4c323bcd03ea4cd768c4ce46f1e786aa3113bc51; but only when there is transition | [Demo](https://codepen.io/MojtabaSamimi/pen/Exjjqqx?editors=1000).

Reply:
Can I ask why? Did you notice some less-than-ideal behaviour w/o this new `withTransition` logic?

Reply:
[Here](https://codepen.io/MojtabaSamimi/pen/VwLvwyq?editors=1000) when applying the very first commit of this PR, the bars were disappeared after third zoom out (on the graphs on the right).

Reply:
Ok great, can you try to lock that down in a test?

Reply:
... and can you add a comment above:

```js
            if(isBlank && withTransition) {
                if(isHorizontal) {
                    x1 = x0;
                } else {
                    y1 = y0;
                }
            }

```

Reply:
> Ok great, can you try to lock that down in a test?

That one was added in  https://github.com/plotly/plotly.js/pull/4568/commits/7f3fb34fc3300f7670a9d92e0054802efb105ac2.

Reply:
Hmm. I don't see a test locking down your https://github.com/plotly/plotly.js/pull/4568/commits/4c323bcd03ea4cd768c4ce46f1e786aa3113bc51 commit.

Reply:
> ... and can you add a comment above:
> 
> ```js
>             if(isBlank && withTransition) {
>                 if(isHorizontal) {
>                     x1 = x0;
>                 } else {
>                     y1 = y0;
>                 }
>             }
> ```

Done in https://github.com/plotly/plotly.js/pull/4568/commits/8ebc0301f1eaf3f4e54eeebab7923bc2bf6d96be.

Reply:
> Hmm. I don't see a test locking down your [4c323bc](https://github.com/plotly/plotly.js/commit/4c323bcd03ea4cd768c4ce46f1e786aa3113bc51) commit.

Added in https://github.com/plotly/plotly.js/pull/4568/commits/494775bfccc1e517917ecabb7cb494c3412d9e53.